components/button/index.js

```diff
@@ -13,11 +13,13 @@ function Button( { href, isPrimary, isLarge, isToggled, className, ...additional
 	} );
 
 	const tag = href !== undefined ? 'a' : 'button';
-	const tagProps = tag === 'a' ? { href } : { type: 'button' };
+	const tagProps = tag === 'a' ? { href, target } : { type: 'button', disabled };
+	const disabledLinkProps = tag === 'a' && !! disabled ? { 'aria-disabled': true, tabIndex: '-1' } : {};
```

Comment:
Yes, I start to think we will never have this use-case but we should do what you suggest as a fallback.